pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/ScalarFunctions.java

```diff
@@ -56,9 +57,9 @@ public static byte[] stPoint(double x, double y) {
    * @return the created point
    */
   @ScalarFunction(names = {"stPoint", "ST_point"})
-  public static byte[] stPoint(double x, double y, boolean isGeography) {
+  public static byte[] stPoint(double x, double y, Object isGeography) {
     Point point = GeometryUtils.GEOMETRY_FACTORY.createPoint(new Coordinate(x, y));
-    if (isGeography) {
+    if (BooleanUtils.toBoolean(isGeography)) {
```

Comment:
what if isGeography is not given boolean up?

Reply:
`BooleanUtils.toBoolean` actually works for number/string/boolean type.

Reply:
I see. for non-boolean value, will it handle exception?

Reply:
this is our internal boolean util:
```
public static boolean toBoolean(Object booleanObject) {
    if (booleanObject == null) {
      return false;
    } else if (booleanObject instanceof String) {
      return BooleanUtils.toBoolean((String) booleanObject);
    } else if (booleanObject instanceof Number) {
      return ((Number) booleanObject).intValue() == INTERNAL_TRUE;
    } else if (booleanObject instanceof Boolean) {
      return (boolean) booleanObject;
    } else {
      throw new IllegalArgumentException("Illegal type for boolean conversion: " + booleanObject.getClass());
    }
  }
```